### DISCUSSION TITLE
raw_window_handle::HasRawWindowHandle not implemented

### DISCUSSION THREAD

**Discussion by ElhamAryanpur** [gfx-rs/wgpu#4989] [Q&A] [answered]
I have been trying to update the version of wgpu in my engine to 0.18 ever since it came out, yet I still cannot solve an issue on [surface creation](https://github.com/AryanpurTech/BlueEngine/blob/9f10faf19b5d35089b51936602e0d8447f8614c8/src/render.rs#L45). I've checked the examples, other people's updates, everywhere and still couldn't understand what caused the issue...

The error I get:
```
the trait bound `winit::window::Window: raw_window_handle::HasRawWindowHandle` is not satisfied
the trait `raw_window_handle::HasRawWindowHandle` is implemented for `&'a T`
required for `&winit::window::Window` to implement `raw_window_handle::HasRawWindowHandle`
```

my dependency:
winit = { version = "0.29" }
wgpu = { version = "0.18" }

[reference I checked from wgpu repo](https://github.com/gfx-rs/wgpu/blob/trunk/examples/src/hello_triangle/mod.rs)
[reference I checked from sotrh's wgpu tutorial repo](https://github.com/sotrh/learn-wgpu/blob/master/code/beginner/tutorial3-pipeline/src/lib.rs)

**Comment by bwoods** [comment, marked as answer]
As `winit` has support for different versions of the `raw_window_handle` protocols, you will need to enable the appropriate version for `wgpu`. Here's what I use:

```toml
# Cargo.toml
winit = { version = "0.29" , features = ["rwh_05"] }
wgpu = "0.18"
```

**Comment by ElhamAryanpur** [comment]
Oh thanks! I read that in an issue somewhere in wgpu too, however when I looked at the wgpu repo's Cargo.toml, it was rwh6 so I thought maybe that's not the issue and moved on 😅
